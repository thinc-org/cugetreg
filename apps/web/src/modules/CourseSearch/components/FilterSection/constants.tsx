import { CreateCheckbox } from '@web/modules/CourseSearch/components/FilterSection/hooks/useFilterBar'

import { DayOfWeek, GenEdType } from '@libs/codegen'
import { DayChipKey, GenEdChipKey } from '@libs/react-ui'

export const createGenEdCheckboxes: CreateCheckbox<GenEdChipKey>[] = [
  {
    label: 'หมวดวิทย์',
    value: GenEdType.Sc,
  },
  {
    label: 'หมวดสังคม',
    value: GenEdType.So,
  },
  {
    label: 'หมวดมนุษย์',
    value: GenEdType.Hu,
  },
  {
    label: 'หมวดสหฯ',
    value: GenEdType.In,
  },
  {
    label: 'ไม่ใช่ GenEd',
    value: GenEdType.No,
  },
]

export const createDayOfWeekCheckboxes: CreateCheckbox<DayChipKey>[] = [
  {
    label: 'วันจันทร์',
    value: DayOfWeek.Mo,
  },
  {
    label: 'วันอังคาร',
    value: DayOfWeek.Tu,
  },
  {
    label: 'วันพุธ',
    value: DayOfWeek.We,
  },
  {
    label: 'วันพฤหัสบดี',
    value: DayOfWeek.Th,
  },
  {
    label: 'วันศุกร์',
    value: DayOfWeek.Fr,
  },
  {
    label: 'วันเสาร์',
    value: DayOfWeek.Sa,
  },
  {
    label: 'วันอาทิตย์',
    value: DayOfWeek.Su,
  },
]
