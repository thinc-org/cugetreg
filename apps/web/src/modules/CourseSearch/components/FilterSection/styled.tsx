import {
  Button as MuiButton,
  DialogContent as MuiDialogContent,
  Paper,
  css,
  styled,
} from '@mui/material'

export const DialogContent = styled(MuiDialogContent)`
  padding: ${({ theme }) => theme.spacing(4)};
`

export const Box = styled('div')`
  display: flex;
  justify-content: space-between;
  div:first-of-type {
    ${({ theme }) => {
      return css`
        margin-right: ${theme.spacing(12)};
        ${theme.breakpoints.down('sm')} {
          margin-right: ${theme.spacing(2)};
        }
      `
    }};
  }
`

export const StickyPaper = styled(Paper)`
  display: flex;
  position: sticky;
  width: fit-content;
  height: fit-content;
  top: 100px;
`

export const Button = styled(MuiButton)`
  margin-top: ${({ theme }) => theme.spacing(2)};
`
