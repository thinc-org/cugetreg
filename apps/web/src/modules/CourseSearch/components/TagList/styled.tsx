import { Stack, styled } from '@mui/material'

export const StyledStack = styled(Stack)`
  height: 24px;
  display: flex;
  justify-content: center;
`
