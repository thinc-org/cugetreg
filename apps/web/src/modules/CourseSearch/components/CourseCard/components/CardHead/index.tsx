import { useTranslation } from 'react-i18next'

import KeyboardArrowDown from '@mui/icons-material/KeyboardArrowDown'
import KeyboardArrowUp from '@mui/icons-material/KeyboardArrowUp'
import { CardHeader, Grid, IconButton, Link, Stack, Typography } from '@mui/material'

import { Analytics } from '@web/common/context/Analytics/components/Analytics'
import { LinkWithAnalytics } from '@web/common/context/Analytics/components/LinkWithAnalytics'
import { COURSE_TITLE, EXPAND_BUTTON } from '@web/common/context/Analytics/constants'
import { useLinkBuilderWithCourseGroup } from '@web/common/hooks/useLinkBuilder'
import { useCourseCard } from '@web/modules/CourseSearch/components/CourseCard/context'

import { GenEdChip } from '@libs/react-ui'

export function CardHead() {
  const { t } = useTranslation('courseCard')
  const { course, isGenEd, isExpanded, onToggle } = useCourseCard()
  const { buildLink } = useLinkBuilderWithCourseGroup(course)

  return (
    <CardHeader
      sx={{ p: 4, pb: 0, pt: 3 }}
      title={
        <Stack direction="row" justifyContent="space-between">
          <Grid container spacing={1}>
            <Grid item>
              <LinkWithAnalytics
                href={buildLink(`/courses/${course.courseNo}`)}
                passHref
                elementName={COURSE_TITLE}
                elementId={course.courseNo}
              >
                <Typography variant="h5" color="primaryRange.500" component={Link}>
                  {course.courseNo} {course.abbrName}
                </Typography>
              </LinkWithAnalytics>
            </Grid>
            <Grid item>
              <Typography variant="h6" color="primaryRange.100">
                {t('credit', { credit: course.credit })}
              </Typography>
            </Grid>
            {isGenEd && (
              <Grid item>
                <GenEdChip
                  sx={{ display: { xs: 'none', sm: 'inline-flex' } }}
                  type={course.genEdType}
                />
              </Grid>
            )}
          </Grid>
          {/* rating is not implemented yet */}
          {/* {isGenEd && (
            <Stack direction="row" spacing={0.25} sx={{ display: { xs: 'none', sm: 'flex' } }}>
              <Star />
              <Typography variant="h6">{rating}</Typography>
            </Stack>
          )} */}
        </Stack>
      }
      action={
        <Analytics elementName={EXPAND_BUTTON} elementId={course.courseNo}>
          <IconButton onClick={onToggle} color="primary">
            {isExpanded ? <KeyboardArrowUp /> : <KeyboardArrowDown />}
          </IconButton>
        </Analytics>
      }
    />
  )
}
