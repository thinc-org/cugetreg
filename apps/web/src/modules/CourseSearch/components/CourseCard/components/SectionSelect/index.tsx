import { FormControl, MenuItem, Select, useMediaQuery, useTheme } from '@mui/material'
import { Analytics } from '@web/common/context/Analytics/components/Analytics'
import { SECTION_SELECT } from '@web/common/context/Analytics/constants'
import { useCourseCard } from '@web/modules/CourseSearch/components/CourseCard/context'

export function SectionSelect() {
  const theme = useTheme()
  const isDesktop = useMediaQuery(theme.breakpoints.up('sm'))
  const { selectedSectionNumber, setSectionNumber, sectionNumbers, course } = useCourseCard()
  return (
    <FormControl fullWidth={isDesktop}>
      <Analytics elementName={SECTION_SELECT} elementId={course.courseNo}>
        {({ log }) => (
          <Select
            value={selectedSectionNumber}
            onChange={(e) => {
              const sectionNumber = e.target.value as string
              log(null, sectionNumber)
              setSectionNumber(sectionNumber)
            }}
            name="sectionNo"
          >
            {sectionNumbers.map((value) => (
              <MenuItem key={value} value={value}>
                Sec {value}
              </MenuItem>
            ))}
          </Select>
        )}
      </Analytics>
    </FormControl>
  )
}
