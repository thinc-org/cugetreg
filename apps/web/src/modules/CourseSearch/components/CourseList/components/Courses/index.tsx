import React from 'react'

import { Typography } from '@mui/material'
import { getCourseKeyString } from '@web/common/utils/getCourseKeyString'
import { CourseCard } from '@web/modules/CourseSearch/components/CourseCard'

import { Course } from '@cgr/codegen'

export interface CoursesProps {
  loading: boolean
  courses: Course[]
}

export const Courses: React.FC<CoursesProps> = ({ courses, loading }) => {
  if (!courses.length) {
    if (loading) return null

    return (
      <>
        <Typography align="center" variant="h5" mt={4}>
          ไม่พบรายวิชา
        </Typography>
        <Typography align="center" variant="subtitle1">
          ลองเปลี่ยนภาคเรียนที่มุมบนขวา
        </Typography>
      </>
    )
  }

  return (
    <>
      {courses.map((course) => (
        <CourseCard key={getCourseKeyString(course)} course={course} />
      ))}
    </>
  )
}
