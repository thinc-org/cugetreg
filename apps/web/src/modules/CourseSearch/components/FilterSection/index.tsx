import { DialogContent, Stack, useTheme } from '@mui/material'
import useMediaQuery from '@mui/material/useMediaQuery'
import useGoogleOptimize from '@react-hook/google-optimize'

import { DayChipKey, GenEdChipKey } from '@web/common/components/Chips/config'
import { ResponsiveDialog } from '@web/common/components/ResponsiveDialog'
import { Analytics } from '@web/common/context/Analytics/components/Analytics'
import {
  DAY_FILTER,
  GENED_FILTER,
  PERIOD_RANGE_FILTER,
} from '@web/common/context/Analytics/constants'
import { GOOGLE_OPTIMIZA_FILTER_ORDER } from '@web/env'
import { CheckboxGroup } from '@web/modules/CourseSearch/components/CheckboxGroup'
import {
  createDayOfWeekCheckboxes, // createSpecialCheckboxes,
  createGenEdCheckboxes,
} from '@web/modules/CourseSearch/components/FilterSection/constants'
import { FilterSectionProps } from '@web/modules/CourseSearch/components/FilterSection/types'
import { tail } from '@web/utils/tail'

import { useHasTags } from '../TagList'
import { SelectTime } from './components/SelectTime'
import { useFilterBar } from './hooks/useFilterBar'
import { Button, StickyPaper } from './styled'

export const FilterSection: React.FC<FilterSectionProps> = ({ open, handleClose }) => {
  const { checkboxes: genEdCheckboxes } = useFilterBar<GenEdChipKey>(
    createGenEdCheckboxes,
    'genEdTypes'
  )
  const { checkboxes: dayOfWeekCheckboxes } = useFilterBar<DayChipKey>(
    createDayOfWeekCheckboxes,
    'dayOfWeeks'
  )
  // const { checkboxes: specialCheckboxes } = useFilterBar(createSpecialCheckboxes)

  const hasTags = useHasTags()

  const theme = useTheme()
  const match = useMediaQuery(theme.breakpoints.up('sm'))
  const isExperimentOrder = useGoogleOptimize(GOOGLE_OPTIMIZA_FILTER_ORDER, [false, true])

  const filters = [
    <Analytics key={1} elementName={GENED_FILTER}>
      {({ log }) => (
        <CheckboxGroup
          log={log}
          id="genEdFilter"
          title="หมวดหมู่ GenEd"
          checkboxes={genEdCheckboxes}
        />
      )}
    </Analytics>,
    <Analytics key={2} elementName={DAY_FILTER}>
      {({ log }) => (
        <CheckboxGroup
          log={log}
          id="dayOfWeekFilter"
          title="วันในสัปดาห์"
          checkboxes={dayOfWeekCheckboxes}
        />
      )}
    </Analytics>,
    <Analytics key={3} elementName={PERIOD_RANGE_FILTER}>
      {({ log }) => <SelectTime log={log} />}
    </Analytics>,
  ]
  const orderedFilters = isExperimentOrder ? [...tail(filters), filters[0]] : filters

  return match ? (
    open ? (
      <StickyPaper hasTags={hasTags} variant="outlined">
        <Stack spacing={4} p={4} pr={2} overflow="auto">
          {orderedFilters}
          {/* <CheckboxGroup title="แสดงผลพิเศษ" checkboxes={specialCheckboxes} /> */}
        </Stack>
      </StickyPaper>
    ) : null
  ) : (
    <ResponsiveDialog open={open} onClose={handleClose} fullWidth>
      <DialogContent>
        <Stack>
          <Stack direction="row" justifyContent="space-between">
            <Analytics elementName={GENED_FILTER}>
              {({ log }) => (
                <CheckboxGroup
                  id="genEdFilter"
                  log={log}
                  title="หมวดหมู่ GenEd"
                  checkboxes={genEdCheckboxes}
                />
              )}
            </Analytics>
            <Analytics elementName={DAY_FILTER}>
              {({ log }) => (
                <CheckboxGroup
                  id="dayOfWeekFilter"
                  log={log}
                  title="วันในสัปดาห์"
                  checkboxes={dayOfWeekCheckboxes}
                />
              )}
            </Analytics>
          </Stack>
          <Analytics elementName={PERIOD_RANGE_FILTER}>
            {({ log }) => <SelectTime log={log} />}
          </Analytics>
          {/* <CheckboxGroup title="แสดงผลพิเศษ" checkboxes={specialCheckboxes} /> */}
        </Stack>
        <Button color="primary" variant="outlined" fullWidth onClick={handleClose}>
          เลือกตัวกรอง
        </Button>
      </DialogContent>
    </ResponsiveDialog>
  )
}
