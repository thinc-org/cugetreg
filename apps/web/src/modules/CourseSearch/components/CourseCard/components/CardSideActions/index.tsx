import { Collapse, Stack } from '@mui/material'
import { SelectButton } from '@web/common/components/SelectButton'
import { Analytics } from '@web/common/context/Analytics/components/Analytics'
import { SUBJECT_SELECT_BUTTON } from '@web/common/context/Analytics/constants'
import { SectionSelect } from '@web/modules/CourseSearch/components/CourseCard/components/SectionSelect'
import { useCourseCard } from '@web/modules/CourseSearch/components/CourseCard/context'

export function CardSideActions() {
  const { course, selectedSectionNumber, isExpanded } = useCourseCard()
  return (
    <Stack
      display={['none', 'flex']}
      flexBasis="auto"
      justifyContent="space-between"
      alignItems="stretch"
      ml={2}
    >
      <Collapse in={isExpanded}>
        <SectionSelect />
      </Collapse>
      <Analytics elementName={SUBJECT_SELECT_BUTTON} elementId={course.courseNo}>
        {({ log }) => (
          <SelectButton
            course={course}
            selectedSectionNumber={selectedSectionNumber}
            log={() => log(null, selectedSectionNumber)}
          />
        )}
      </Analytics>
    </Stack>
  )
}
