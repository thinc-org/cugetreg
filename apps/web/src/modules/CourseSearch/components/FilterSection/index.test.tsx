import { ResponsiveDialog } from '@web/common/components/ResponsiveDialog'
import { Analytics } from '@web/common/context/Analytics/components/Analytics'
import { DAY_FILTER, GENED_FILTER } from '@web/common/context/Analytics/constants'
import { shallow } from 'enzyme'

import { DayOfWeek, GenEdType } from '@cgr/codegen'

import { Button, StickyPaper } from './styled'

describe('FilterSection', () => {
  const MOCK_HASTAG = true
  const MOCK_QUERY = 'query'

  const mockUseMediaQuery = jest.fn()
  const mockUseFilterBar = jest.fn(() => ({
    checkboxes: [],
  }))
  const mockUseHasTags = jest.fn(() => MOCK_HASTAG)
  const mockHandleClose = jest.fn()
  const mockUseTheme = jest.fn(() => ({
    breakpoints: {
      up: jest.fn(() => MOCK_QUERY),
    },
  }))

  jest.doMock('@mui/material/useMediaQuery', () => mockUseMediaQuery)
  jest.doMock('./hooks/useFilterBar', () => ({ useFilterBar: mockUseFilterBar }))
  jest.doMock('../TagList', () => ({ useHasTags: mockUseHasTags }))
  jest.doMock('@mui/material', () => ({
    ...(jest.requireActual('@mui/material') as Record<string, unknown>),
    useTheme: mockUseTheme,
  }))

  afterEach(() => {
    jest.clearAllMocks()
  })

  it.each`
    matchSmallScreen | isExpandFilter
    ${true}          | ${true}
    ${true}          | ${false}
    ${false}         | ${true}
    ${false}         | ${false}
  `(
    'Should matched snapshot correctly when matchSmallScreen=$matchSmallScreen and isExpandFilter=$isExpandFilter',
    async ({ matchSmallScreen, isExpandFilter }) => {
      const { FilterSection } = await import('.')
      const mockHandleClose = jest.fn()
      mockUseMediaQuery.mockReturnValue(matchSmallScreen)

      const wrapper = shallow(<FilterSection open={isExpandFilter} handleClose={mockHandleClose} />)

      expect(mockUseMediaQuery).toBeCalledWith(MOCK_QUERY)
      expect(mockUseTheme).toBeCalledTimes(1)
      expect(mockUseHasTags).toBeCalledTimes(1)
      expect(mockUseFilterBar).toBeCalledTimes(2)
      expect(mockUseFilterBar).toHaveBeenNthCalledWith(
        1,
        [
          {
            label: 'หมวดวิทย์',
            value: GenEdType.Sc,
          },
          {
            label: 'หมวดสังคม',
            value: GenEdType.So,
          },
          {
            label: 'หมวดมนุษย์',
            value: GenEdType.Hu,
          },
          {
            label: 'หมวดสหฯ',
            value: GenEdType.In,
          },
          {
            label: 'ไม่ใช่ GenEd',
            value: GenEdType.No,
          },
        ],
        'genEdTypes'
      )
      expect(mockUseFilterBar).toHaveBeenNthCalledWith(
        2,
        [
          {
            label: 'วันจันทร์',
            value: DayOfWeek.Mo,
          },
          {
            label: 'วันอังคาร',
            value: DayOfWeek.Tu,
          },
          {
            label: 'วันพุธ',
            value: DayOfWeek.We,
          },
          {
            label: 'วันพฤหัสบดี',
            value: DayOfWeek.Th,
          },
          {
            label: 'วันศุกร์',
            value: DayOfWeek.Fr,
          },
          {
            label: 'วันเสาร์',
            value: DayOfWeek.Sa,
          },
          {
            label: 'วันอาทิตย์',
            value: DayOfWeek.Su,
          },
        ],
        'dayOfWeeks'
      )

      if (!matchSmallScreen) {
        expect(wrapper.find(ResponsiveDialog).prop('onClose')).toBe(mockHandleClose)
        expect(wrapper.find(Button).prop('onClick')).toBe(mockHandleClose)
      }
      if (matchSmallScreen && isExpandFilter) {
        expect(wrapper.find(StickyPaper).prop('hasTags')).toBe(MOCK_HASTAG)
      }
      expect(wrapper).toMatchSnapshot()
    }
  )

  it.each`
    matchSmallScreen | isExpandFilter
    ${true}          | ${true}
    ${false}         | ${true}
    ${false}         | ${false}
  `(
    'Should render CheckboxGroup of GenEds correctly when when matchSmallScreen=$matchSmallScreen and isExpandFilter=$isExpandFilter',
    async ({ matchSmallScreen, isExpandFilter }) => {
      const { FilterSection } = await import('.')
      mockUseMediaQuery.mockReturnValue(matchSmallScreen)

      const wrapper = shallow(<FilterSection open={isExpandFilter} handleClose={mockHandleClose} />)
        .find(Analytics)
        .findWhere((n) => n.prop('elementName') === GENED_FILTER)
        .renderProp('children')({ log: () => {} })
      expect(wrapper).toMatchSnapshot()
    }
  )

  it.each`
    matchSmallScreen | isExpandFilter
    ${true}          | ${true}
    ${false}         | ${true}
    ${false}         | ${false}
  `(
    'Should render CheckboxGroup of days correctly when matchSmallScreen=$matchSmallScreen and isExpandFilter=$isExpandFilter',
    async ({ matchSmallScreen, isExpandFilter }) => {
      const { FilterSection } = await import('.')
      mockUseMediaQuery.mockReturnValue(matchSmallScreen)

      const wrapper = shallow(<FilterSection open={isExpandFilter} handleClose={mockHandleClose} />)
        .find(Analytics)
        .findWhere((n) => n.prop('elementName') === DAY_FILTER)
        .renderProp('children')({ log: () => {} })
      expect(wrapper).toMatchSnapshot()
    }
  )
})
