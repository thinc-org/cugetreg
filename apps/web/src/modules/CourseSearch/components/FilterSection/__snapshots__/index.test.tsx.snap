// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSection Should matched snapshot correctly when matchSmallScreen=false and isExpandFilter=false 1`] = `
<ResponsiveDialog
  fullWidth={true}
  onClose={[MockFunction]}
  open={false}
>
  <ForwardRef(DialogContent)>
    <ForwardRef(Stack)>
      <ForwardRef(Stack)
        direction="row"
        justifyContent="space-between"
      >
        <Analytics
          elementName="gened_search_filter"
        >
          <Component />
        </Analytics>
        <Analytics
          elementName="day_search_filter"
        >
          <Component />
        </Analytics>
      </ForwardRef(Stack)>
      <Analytics
        elementName="period_range_filter"
      >
        <Component />
      </Analytics>
    </ForwardRef(Stack)>
    <Styled(ForwardRef(Button))
      color="primary"
      fullWidth={true}
      onClick={[MockFunction]}
      variant="outlined"
    >
      เลือกตัวกรอง
    </Styled(ForwardRef(Button))>
  </ForwardRef(DialogContent)>
</ResponsiveDialog>
`;

exports[`FilterSection Should matched snapshot correctly when matchSmallScreen=false and isExpandFilter=true 1`] = `
<ResponsiveDialog
  fullWidth={true}
  onClose={[MockFunction]}
  open={true}
>
  <ForwardRef(DialogContent)>
    <ForwardRef(Stack)>
      <ForwardRef(Stack)
        direction="row"
        justifyContent="space-between"
      >
        <Analytics
          elementName="gened_search_filter"
        >
          <Component />
        </Analytics>
        <Analytics
          elementName="day_search_filter"
        >
          <Component />
        </Analytics>
      </ForwardRef(Stack)>
      <Analytics
        elementName="period_range_filter"
      >
        <Component />
      </Analytics>
    </ForwardRef(Stack)>
    <Styled(ForwardRef(Button))
      color="primary"
      fullWidth={true}
      onClick={[MockFunction]}
      variant="outlined"
    >
      เลือกตัวกรอง
    </Styled(ForwardRef(Button))>
  </ForwardRef(DialogContent)>
</ResponsiveDialog>
`;

exports[`FilterSection Should matched snapshot correctly when matchSmallScreen=true and isExpandFilter=false 1`] = `""`;

exports[`FilterSection Should matched snapshot correctly when matchSmallScreen=true and isExpandFilter=true 1`] = `
<Styled(ForwardRef(Paper))
  hasTags={true}
  variant="outlined"
>
  <ForwardRef(Stack)
    overflow="auto"
    p={4}
    pr={2}
    spacing={4}
  >
    <Analytics
      elementName="gened_search_filter"
      key="1"
    >
      <Component />
    </Analytics>
    <Analytics
      elementName="day_search_filter"
      key="2"
    >
      <Component />
    </Analytics>
    <Analytics
      elementName="period_range_filter"
      key="3"
    >
      <Component />
    </Analytics>
  </ForwardRef(Stack)>
</Styled(ForwardRef(Paper))>
`;

exports[`FilterSection Should render CheckboxGroup of GenEds correctly when when matchSmallScreen=false and isExpandFilter=false 1`] = `
<CheckboxGroup
  checkboxes={Array []}
  id="genEdFilter"
  log={[Function]}
  title="หมวดหมู่ GenEd"
/>
`;

exports[`FilterSection Should render CheckboxGroup of GenEds correctly when when matchSmallScreen=false and isExpandFilter=true 1`] = `
<CheckboxGroup
  checkboxes={Array []}
  id="genEdFilter"
  log={[Function]}
  title="หมวดหมู่ GenEd"
/>
`;

exports[`FilterSection Should render CheckboxGroup of GenEds correctly when when matchSmallScreen=true and isExpandFilter=true 1`] = `
<CheckboxGroup
  checkboxes={Array []}
  id="genEdFilter"
  log={[Function]}
  title="หมวดหมู่ GenEd"
/>
`;

exports[`FilterSection Should render CheckboxGroup of days correctly when matchSmallScreen=false and isExpandFilter=false 1`] = `
<CheckboxGroup
  checkboxes={Array []}
  id="dayOfWeekFilter"
  log={[Function]}
  title="วันในสัปดาห์"
/>
`;

exports[`FilterSection Should render CheckboxGroup of days correctly when matchSmallScreen=false and isExpandFilter=true 1`] = `
<CheckboxGroup
  checkboxes={Array []}
  id="dayOfWeekFilter"
  log={[Function]}
  title="วันในสัปดาห์"
/>
`;

exports[`FilterSection Should render CheckboxGroup of days correctly when matchSmallScreen=true and isExpandFilter=true 1`] = `
<CheckboxGroup
  checkboxes={Array []}
  id="dayOfWeekFilter"
  log={[Function]}
  title="วันในสัปดาห์"
/>
`;
